vcs/src/test/java/org/openjdk/skara/vcs/RepositoryTests.java

```diff
@@ -742,7 +742,6 @@ void testNonCheckedOutRepositoryIsHealthy(VCS vcs) throws IOException {
     void testBranchesOnEmptyRepository(VCS vcs) throws IOException {
         try (var dir = new TemporaryDirectory()) {
             var r = TestableRepository.init(dir.path(), vcs);
-            var expected = vcs == VCS.GIT ? List.of() : List.of(new Branch("default"));
             assertEquals(List.of(), r.branches());
```

Comment:
I am just curious about the expected behavior when testing on a mercurial (hg) repository.

When I initialize a new hg repository and run `hg branch`, it returns "default". However, when I use `hg branches`, it returns empty. So is "default" branch ignored?

Reply:
From my experiments, Git (2.45.2) and Mercurial (6.8) behave identically in that regard. Here's a newly initialised Git repo:

    % git branch --list --all
    % git branch --show-current
    master
    %

It's only after I make the first commit to the default branch, be it Git or Mercurial, that branch shows up in the output.